File: app/src/main/java/org/mifos/mobilebanking/utils/MFErrorParser.java
package org.mifos.mobilebanking.utils;

import com.google.gson.Gson;

import org.mifos.mobilebanking.models.mifoserror.MifosError;

import io.reactivex.plugins.RxJavaPlugins;
import retrofit2.HttpException;

public class MFErrorParser {

    public static final String LOG_TAG = "MFErrorParser";

    private static Gson gson = new Gson();

    public static MifosError parseError(String serverResponse) {
        return gson.fromJson(serverResponse, MifosError.class);
    }

    public static String errorMessage(Throwable throwableError) {
        String errorMessage = "";
        try {
            if (throwableError instanceof HttpException) {
                errorMessage = ((HttpException) throwableError).response().errorBody().string();
                errorMessage =  MFErrorParser.parseError(errorMessage).getErrors()
                        .get(0).getDefaultUserMessage();
            }
        } catch (Throwable throwable) {
            RxJavaPlugins.getErrorHandler();
        }
        return errorMessage;
    }
}

File: app/src/main/java/org/mifos/mobile/utils/MFErrorParser.kt
package org.mifos.mobile.utils

import com.google.gson.Gson
import io.reactivex.plugins.RxJavaPlugins
import org.mifos.mobile.models.mifoserror.MifosError
import retrofit2.HttpException

object MFErrorParser {
    const val LOG_TAG = "MFErrorParser"
    private val gson = Gson()
    fun parseError(serverResponse: String?): MifosError {
        return gson.fromJson(serverResponse, MifosError::class.java)
    }

    @kotlin.jvm.JvmStatic
    fun errorMessage(throwableError: Throwable?): String? {
        var errorMessage: String? = ""
        try {
            if (throwableError is HttpException) {
                errorMessage = throwableError.response().errorBody().string()
                errorMessage = parseError(errorMessage).errors[0].defaultUserMessage
            }
        } catch (throwable: Throwable) {
            RxJavaPlugins.getErrorHandler()
        }
        return errorMessage
    }
}